neovim/ui/gtk_ui.py

```diff
@@ -152,7 +152,11 @@ def _nvim_resize(self, columns, rows):
         self._cell_pixel_width = cell_pixel_width
         self._cell_pixel_height = cell_pixel_height
         self._screen = Screen(columns, rows)
-        self._window.resize(pixel_width, pixel_height)
+        prev_rect = self._drawing_area.get_allocation()
```

Comment:
I think that the way to deal with it cleanly, is to draw nvim in a custom widget instead of using DrawingArea. With a custom widget, we can properly reply to """get_preferred_width""" and """get_preferred_height""" etc, and the GTK lib will deal with the rest. After it had figure out the geometry, we can go and render.

There is some info here:
https://developer.gnome.org/gtk3/stable/GtkWidget.html

That would require to split gtk_ui and have another gtk_widget, that would be used by gtk_ui.py. Does it sound good?